manager/drivers/secrets.go

```diff
@@ -26,12 +26,46 @@ func NewSecretDriver(plugin plugingetter.CompatPlugin) *SecretDriver {
 }
 
 // Get gets a secret from the secret provider
-func (d *SecretDriver) Get(spec *api.SecretSpec) ([]byte, error) {
+func (d *SecretDriver) Get(spec *api.SecretSpec, task *api.Task) ([]byte, error) {
 	if spec == nil {
 		return nil, fmt.Errorf("spec is nil")
 	}
+	if task == nil {
+		return nil, fmt.Errorf("service spec is nil")
+	}
+
+	container := task.Spec.GetContainer()
+	var hostname string
```

Comment:
minor nit: The `hostname` variable isn't really necessary. We could move the `if container != nil` below the initialization of `secretReq` and fill in `ServiceHostname` directly.